homeassistant/components/apsystems/__init__.py

```diff
@@ -12,15 +14,23 @@
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
-ApsystemsConfigEntry = ConfigEntry[ApSystemsDataCoordinator]
+ApsystemsConfigEntry = ConfigEntry["ApSystemsData"]
+
+
+@dataclass
+class ApSystemsData:
+    """Store runtime data."""
+
+    coordinator: ApSystemsDataCoordinator
+    api: APsystemsEZ1M
```

Comment:
Like @joostlek, I don't think you should introduce the `api` in this PR. You should only introduce it when you start making use of it.

However, I do think you could add the `device_id` to the `ApSystemsData`, so that you only have the assert once in the primary `async_setup_entry`.
Then you have it available already for all subsequent uses.

```suggestion
    device_id: str
```